### DISCUSSION TITLE
Interpolation method in SolutionFunction

### DISCUSSION THREAD

**Discussion by hsheldon** [idaholab/moose#18026] [Q&A General] [answered]
I am trying to understand how information is interpolated from a UserObject onto a new mesh when using a SolutionFunction or SolutionAux. The documentation for SolutionAux states that “a point locator is used to find the nearest node or element and the corresponding value.”  This seems to suggest that no interpolation is performed; instead the value at the nearest node or element is used. Is this correct?  If so, is there an alternative method that does actually interpolate between points in the UserObject mesh to determine the values at points in the new mesh?
There is not much documentation for SolutionFunction but looking at the code it seems to do essentially the same thing as SolutionAux.

**Comment by GiudGiud** [comment, marked as answer]
Hello

I went to check in SolutionUserObject and libmesh::MeshFunction is used to interpolate the solutions. So more information / the code doing the interpolation can be found there: https://libmesh.github.io/doxygen/classlibMesh_1_1MeshFunction.html#ae1c64240672461d67f43abba17a5cb18

It can also do interpolation in time btw. 

Best,
Guillaume

**Reply by hsheldon** [reply]
Ok that's good news, so it does actually interpolate. Would be good to update the doco for SolutionAux to clarify.
